scRNA-seq-advanced/setup/intro-setup.Rmd

```diff
@@ -0,0 +1,272 @@
+  dplyr::filter(Type == "Antibody Capture") %>% 
+  dplyr::pull(ID)
+
+adt_pbmc <- pbmc_sce_percellqc[rownames(pbmc_sce_percellqc) %in% adt_names]
+
+# split out the experiments:
+pbmc_sce_percellqc <- pbmc_sce_percellqc[rownames(pbmc_sce_percellqc) %in% gene_names]
+altExp(pbmc_sce_percellqc, "CITEseq") <- adt_pbmc
+```
+
+
+
+### Filter RNA
+
+#### First, with `DropletUtils`
+
+```{r}
+# Filter RNA
+droplet_df <- DropletUtils::emptyDrops(
+  counts(pbmc_sce_percellqc), 
+  BPPARAM = BiocParallel::MulticoreParam(4)
+) 
+
+# Filter down to low FDR cells and check out dimensions
+# Values here are mostly NA.
+# NOTE: this filtering does propagate to altExp!! :) 
+cells_to_retain <- which(droplet_df$FDR <= 0.01)
+filtered_sce <- pbmc_sce_percellqc[, cells_to_retain]
+dim(filtered_sce) 
+```
+
+
+#### We can also filter RNA with `miQC`
+
+Mito QC stats are already present in `pbmc_sce_percellqc`.
+This is sometimes working and sometimes not for me, so commenting out for now...
+```{r}
+#miqc_model <- miQC::mixtureModel(pbmc_sce_percellqc) 
+#filtered_sce_miqc <- miQC::filterCells(pbmc_sce_percellqc,
+#                                       model = miqc_model)
+#
+#dim(filtered_sce_miqc)
+```
+Moving forward, one can choose to proceed with either `filtered_sce_miqc` or `filtered_sce`; we'll use `filtered_sce` for now.
+
+
+#### Compare `DropletUtils` to `miQC` filtering
+
+We may want to compare certain aspects of cells that were filtered out with DropletUtils vs miQC.
+Chunk below, currently pending content..., can do that!
+
+```{r}
+
+```
+
+### Filter ADT
+
+This data does have a control antibody (IgG1), so we can leverage this ADT for QC and filtering.
+```{r}
+# Filter ADT
+# http://bioconductor.org/books/3.16/OSCA.advanced/integrating-with-protein-abundance.html#applying-custom-qc-filters
+
+controls <- grep("^Ig", rownames(altExp(filtered_sce)))
+adt_qc_df <- DropletUtils::cleanTagCounts(altExp(filtered_sce), controls=controls) # I feel like this control might be crap!
+
+summary(adt_qc_df$high.controls) # these are all NA????????
+summary(adt_qc_df$discard) # these are MOSTLY NA????????
+
+# Let's do this - remove cells that have zero ambient concentrations:
+# This removes 11 cells
+cells_to_retain <- which(adt_qc_df$zero.ambient == FALSE)
+filtered_sce <- filtered_sce[, cells_to_retain]
+dim(filtered_sce)
+```
+
+
+
+### Normalize RNA and ADT:
+
+
+```{r}
+# Normalize RNA
+qclust <- scran::quickCluster(filtered_sce)
+filtered_sce <- scran::computeSumFactors(filtered_sce, clusters = qclust)
+norm_sce <- scater::logNormCounts(filtered_sce)
+
+# Normalize ADT
+
+# Calculate baseline:
+# Try out control-based normalization here 
+#  http://bioconductor.org/books/3.16/OSCA.advanced/integrating-with-protein-abundance.html#control-based-normalization
+control_sf <- scater::librarySizeFactors(altExp(norm_sce), subset_row=controls)  # `controls` was previously defined during filtering
+sum(control_sf == 0) # _HOLY WOW_
```

Comment:
We may not want to get into this in this notebook; I want to show the basics of ADT reading here, but not necessarily spend much time on the interpretation. We aren't going to do integration between RNA & ADT, I don't expect.